modules/core/05-port/types/module.go

```diff
@@ -140,6 +140,9 @@ type UpgradableModule interface {
 		ctx sdk.Context,
 		portID,
 		channelID string,
+		order channeltypes.Order,
+		connectionHops []string,
+		version string,
```

Comment:
The spec doesn't include these parameters in the callback. What's the reason to add them? What checksI  we want to do?

Reply:
In the initial handshake app callbacks, the app is expected to do validation of upgrade params only. So adding them to this callback allows them to get access the fields once again when the upgrade is finalized. 

E.g. in fee middleware, we can only validate version info in the upgrade handshake, and can't set fee enabled or disabled until the upgrade is finalized - otherwise packets recv/ack logic within ics29 would start running in the app stack. 
On upgrade open, the only way we can infer whether to set fee enabled or disabled is by what the final upgrade version looks like.

Reply:
wouldn't it we be able to achieve same effect by having the write for upgrade open be performed before the callback?

Reply:
Yes, in theory, I think suggested that we should do that before as it doesn't really make sense to call the app callback first - same with the `onChanUpgradeRestore` callback. 

However, if you take that approach and remove the arguments, the app would have to manually lookup fields if it needs to do any action which seems like poor application dev ux.